addons/web/static/src/views/relational_model.js

```diff
@@ -2194,9 +2194,17 @@ export class DynamicGroupList extends DynamicList {
     }
 
     async deleteRecords() {
+        const allResIds = [];
         for (const group of this.groups) {
-            group.list.deleteRecords();
-        }
+            const resIds = await group.list.deleteRecords();
+            // Update the count of each group.
+            group.count = group.count - resIds.length;
+            // Collect the deleted resIds from each group.
+            allResIds.push(...resIds);
```

Comment:
how does it behave when grouped on a many2many field, if the same record is deleted from several groups?

Reply:
Apparently, when "group by m2m" was introduced, deleting is not supported.

The following code links to the logic that hides the "Delete" button from the action menu items when the records are grouped by m2m.

https://github.com/odoo/odoo/blob/73af1d04d9d49eace16e3f88c32ce5d457b44b83/addons/web/static/src/views/list/list_controller.js#L271